From: Michael J Rubinsky <mrubinsk@horde.org>
Date: Mon, 31 Oct 2022 23:28:41 -0400
Subject: [PATCH] Avoid access array offset errors

---
 imp-6.2.27/lib/Contents.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/imp-6.2.27/lib/Contents.php b/imp-6.2.27/lib/Contents.php
index a050c4e..9088930 100644
--- a/imp-6.2.27/lib/Contents.php
+++ b/imp-6.2.27/lib/Contents.php
@@ -665,7 +665,7 @@ class IMP_Contents
 
         /* Don't show empty parts. */
         if (($textmode == 'inline') &&
-            !is_null($ret[$mime_id]['data']) &&
+            !empty($ret[$mime_id]) && !is_null($ret[$mime_id]['data']) &&
             !strlen($ret[$mime_id]['data']) &&
             !isset($ret[$mime_id]['status'])) {
             $ret[$mime_id] = null;
